Add blog post for 2.10.0

utkarsharma2 opened the pull request.

utkarsharma2 commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
```diff
@@ -0,0 +1,129 @@
+![Airflow Light mode](airflow_light_mode.png)
+
+## Contributors
+Thanks to everyone who contributed to this release, including Amogh Desai, Andrey Anshin, Brent Bovenzi, Daniel Standish, Ephraim Anierobi, Hussein Awala, Jarek Potiuk, Jed Cunningham, Jens Scheffler, Tzu-ping Chung, Vincent Beck, Wei Lee, and over 120 others!
```
This is just a placeholder from the last post. I need to figure out how to get this.

utkarsharma2 commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
```diff
@@ -0,0 +1,129 @@
+date: "2024-08-08"
+---
+
+'m happy to announce that Apache Airflow 2.10.0 is now available, bringing an array of noteworthy enhancements and new features that will greatly serve our community.
```
```suggestion
I'm happy to announce that Apache Airflow 2.10.0 is now available, bringing an array of noteworthy enhancements and new features that will greatly serve our community.
```

utkarsharma2 commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
```diff
@@ -0,0 +1,129 @@
+
+I'm happy to announce that Apache Airflow 2.10.0 is now available, bringing an array of noteworthy enhancements and new features that will greatly serve our community.
+
+Apache Airflow 2.10.0 contains over 135 commits, which include 43 new features,  85 improvements, 43 bug fixes, and 26 documentation changes.
```
The numbers seem off. I need to figure out how to get the proper data.

jedcunningham replied:
We normally just get the counts from the release notes. Close enough.

utkarsharma2 replied:
@jedcunningham I updated the count using [git compare](https://github.com/apache/airflow/compare/2.9.0...2.10.0rc1). PTAL

jedcunningham replied:
My gut tells me you've swung too far in the other direction. Trying to get just core commits, roughly.

jedcunningham commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
````diff
@@ -0,0 +1,129 @@
+```python
+BashOperator(
+    task_id="hello_world",
+    executor="LocalExecutor",
````
Wouldn't using `ShortName` for these better illustrate that aliases are supported?

o-nikolas replied:
+1

Also LocalExecutor, in this example, is the default executor (first in the list), so it will run any task without a specific executor configured on the task, so specifying it is kind of redundant.

If this is copied from our docs we should fix it there as well!

o-nikolas replied:
I would suggest changing the config to something like:
`'LocalExecutor,KubernetesExecutor,my.custom.module.ExecutorClass:ShortName'`

And then in some examples use `ShortName` and in others use `KubernetesExecutor` so that folks can see the flexibility.

utkarsharma2 replied:
I updated the examples as per the suggestion. PTAL.

jedcunningham commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
```diff
@@ -0,0 +1,129 @@
+
+@task(outlets=[DatasetAlias("my-task-outputs")])
+def my_task_with_outlet_events(*, outlet_events):
+    outlet_events["my-task-outputs"].add(Dataset("s3://bucket/my-task"), extra={"k": "v"})
```
```suggestion
    outlet_events["my-task-outputs"].add(Dataset("s3://bucket/my-task"), extra={"k": "v"})
```

I'd get rid of the extra here, it's just extra noise as it isn't related to DatasetAlias.

jedcunningham commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
````diff
@@ -0,0 +1,129 @@
+There are two options for scheduling based on dataset aliases. Schedule based on `DatasetAlias` or real datasets.
+
+```python
+with DAG(dag_id="dataset-producer"):
+    @task(outlets=[Dataset("example-alias")])
+    def produce_dataset_events():
+        pass
````
Don't need this for the example, right? This is a Dataset for `example-alias`, which isn't an alias (confusing) and also isn't used...

utkarsharma2 replied:
True, removed it.

jedcunningham commented:
Missing TI history!

o-nikolas commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
```diff
@@ -0,0 +1,174 @@
+🚏 Constraints: https://github.com/apache/airflow/tree/constraints-2.10.0
+
+
+## Hybrid Execution
```
```suggestion
## Multiple Executor Configuration (formerly "Hybrid Execution")
```

o-nikolas commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
```diff
@@ -0,0 +1,174 @@
+🚏 Constraints: https://github.com/apache/airflow/tree/constraints-2.10.0
+
+
+## Hybrid Execution
```
I'd also like to write a blog post entirely for multi exec config, since there is some nuance and details there that's worth spelling out and having it on the website blog like [this example](https://airflow.apache.org/blog/introducing_setup_teardown/). I have not created one before, should I just follow this PR as an example of how to do that? I have the content written already but would need to translate it into this formatting.

o-nikolas replied:
CC @jedcunningham

jedcunningham replied:
Sorry, missed this. Yeah. Totally open to that and we can link to it here! We did that for [setup/teardown in 2.7](https://airflow.apache.org/blog/introducing_setup_teardown/), if you need another example.

utkarsharma2 commented:
> Missing TI history!

Added it, PTAL.

vatsrahul1001 commented:
Missing DAG parsing button in UI?

utkarsharma2 commented:
> Missing DAG parsing button in UI?

I added it here - 

> Missing DAG parsing button in UI?

I added it under, `## Additional new features` Wasn't sure how important this feature is from the user's perspective. But I'm open to moving it into its own section.

potiuk commented on landing-pages/site/content/en/blog/airflow-2.10.0/index.md:
```diff
@@ -0,0 +1,182 @@
+* Deferrable operators can now execute directly from the triggerer without needing to go through the worker. This is especially efficient for certain operators, like sensors, and can help teams save both time and money.
+* Airflow 2.10 introduces a new button for on-demand DAG reparsing.
+* Crucial executor logs are now integrated into the task logs. If the executor fails to start a task, the relevant error messages will be available in the task logs, simplifying the debugging process.
+
```
Should we add Open Telemetry traces ? cc : @howardyoo

howardyoo commented:
Hi Jarek,
yes we should!!

Howard

On Tue, Aug 13, 2024 at 6:25 PM Jarek Potiuk ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In landing-pages/site/content/en/blog/airflow-2.10.0/index.md
> <https://github.com/apache/airflow-site/pull/1052#discussion_r1716090675>:
>
> > +
> +![Render dataset conditions in graph view](render_dataset_conditions.png)
> +
> +### Dataset event info in DAG Graph
> +For a DAG run, users can now view the dataset events connected to it directly in the graph view.
> +
> +![Dataset event info](dataset_info.png)
> +
> +## Additional new features
> +
> +Here are just a few interesting new features since there are too many to list in full:
> +
> +* Deferrable operators can now execute directly from the triggerer without needing to go through the worker. This is especially efficient for certain operators, like sensors, and can help teams save both time and money.
> +* Airflow 2.10 introduces a new button for on-demand DAG reparsing.
> +* Crucial executor logs are now integrated into the task logs. If the executor fails to start a task, the relevant error messages will be available in the task logs, simplifying the debugging process.
> +
>
> Should we add Open Telemetry traces ? cc : @howardyoo
> <https://github.com/howardyoo>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/airflow-site/pull/1052#pullrequestreview-2236825411>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHZNLLUS7UNJUGNUJISWIUTZRKIYHAVCNFSM6AAAAABMGUN2W2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDEMZWHAZDKNBRGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

potiuk commented:
> Hi Jarek,
yes we should!!

Maybe you can propose a PR with description/ screenshots @howardyoo ? Also that might be useful segway to explain the OTEL provider capabilities (i.e. how they allow to extend traces).

utkarsharma2 commented:
> Missing DAG parsing button in UI?

@vatsrahul1001 I have moved it to its own section.

utkarsharma2 commented:
> > Hi Jarek,
> > yes we should!!
> 
> Maybe you can propose a PR with description/ screenshots @howardyoo ? Also that might be useful segway to explain the OTEL provider capabilities (i.e. how they allow to extend traces).

Yes, PRs with screenshots would be nice for us to use in the post. I will wait for the same. @howardyoo

vatsrahul1001 commented:
@utkarsharma2 can we document [PR](https://github.com/apache/airflow/pull/39701) as well
